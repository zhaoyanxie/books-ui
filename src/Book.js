import React, { Component } from 'react';
import { API_HOST } from './utils/configVars';

export default class Book extends Component {
	constructor() {
		super();
		this.state = {
			books: []
		};
	}

	async getBooks() {
		const url = `${API_HOST}/books`;
		const response = await fetch(url);
		const data = await response.json();
		if (response.ok) {
			this.setState({
				books: data
			});
		}
	}

	async componentDidMount() {
		await this.getBooks();
	}

	render() {
		return (
			<div className="book">
				<h1>Books</h1>
				{this.state.books.map((book) => {
					return <li key={book._id}>{book.title}</li>;
				})}
			</div>
		);
	}
}
