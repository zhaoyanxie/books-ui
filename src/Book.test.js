import React from 'react';
import Book from './Book';
import ShallowRenderer from 'react-test-renderer/shallow';

test('should render only the book title before componentDidMount is called', () => {
	const renderer = new ShallowRenderer();
	renderer.render(<Book />);
	const output = renderer.getRenderOutput();

	expect(output).toMatchSnapshot();
});

test('should render title and booklist after component did mount', async () => {
	fetch.mockResponseOnce(
		JSON.stringify([
			{
				title: 'book title 1',
				_id: '12345'
			},
			{
				title: 'book title 2',
				_id: '23456'
			}
		])
	);
	const renderer = new ShallowRenderer();
	renderer.render(<Book />);

	const instance = renderer.getMountedInstance();
	await instance.componentDidMount();

	const output = renderer.getRenderOutput();
	expect(output).toMatchSnapshot();
});
