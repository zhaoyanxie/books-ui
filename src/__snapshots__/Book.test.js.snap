// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render only the book title before componentDidMount is called 1`] = `
<div
  className="book"
>
  <h1>
    Books
  </h1>
</div>
`;

exports[`should render title and booklist after component did mount 1`] = `
<div
  className="book"
>
  <h1>
    Books
  </h1>
  <li>
    book title 1
  </li>
  <li>
    book title 2
  </li>
</div>
`;
